firebase-test/firebasetest.js

```diff
@@ -33,3 +48,45 @@ dbUsername.on("value", function(dataSnapshot) {
 	usernameBox.textContent = dataSnapshot.val();
 	console.log( dataSnapshot.val() );
 });
+
+// Event Listener for Login
+// Use Firebase with Github Auth to log in the user
+firebase.auth().signInWithRedirect(provider).catch(function(error){
```

Comment:
I don't see any event listener here though! ***When*** should the user be redirected to log in? What event are you listening for from the user, and ***where*** exactly is that event going to happen on the HTML page? Take a look at previous examples of JavaScript event listeners and use one as a reference.